### DISCUSSION TITLE
Unable to install foundry on windows with WSL

### DISCUSSION THREAD

**Discussion by an-emptymind** [Cyfrin/foundry-full-course-cu#206] [Q&A] [answered]
I am using this command to install foundry on WSL ubuntu terminal
curl -L https://foundry.paradigm.xyz | bash

This is what I am getting :

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105037251/3b299701-090d-4188-a35f-1d46e00e538d)

I have installed WSL,Ubuntu,rust, git all these things. it still doesn't work.
My internet connection is stable & I tried VPN too!
Please help!

**Comment by Shiva-Sai-ssb** [comment, marked as answer]
Hello @Raghavendra-Havale , I encountered a similar problem to yours when attempting to install Foundry on WSL as a Windows user. However, I managed to resolve it by connecting to a VPN. You mentioned that you tried using a VPN, but it wasn't successful, and I'm not aware of the specific VPN software you used. I recommend trying OpenVPN instead, and you can follow this instructional video to set up and establish a connection to the VPN.

https://youtu.be/jiPlad0SJ8c

**Reply by an-emptymind** [reply]
Done bro, Thanks alot it worked!
But there's a new problem.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105037251/f51e84b4-0827-41b2-98b3-c4719c01f5f7)

Foundryup command failed. I've downloaded the said .tar.gz file,
what to do with it now?

**Comment by Shiva-Sai-ssb** [comment]
> Done bro, Thanks alot it worked! But there's a new problem. ![image](https://user-images.githubusercontent.com/105037251/250285464-f51e84b4-0827-41b2-98b3-c4719c01f5f7.png)
> 
> Foundryup command failed. I've downloaded the said .tar.gz file, what to do with it now?

Apologies for the delayed response. I did not encounter any issues during the installation process. Could you please attempt it once more? I will search for a solution to resolve this issue.

**Reply by an-emptymind** [reply]
thanks alot for helping me.
Foundryup command did not work for me. which is why I am trying alternate ways that are given here:
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105037251/9b1e2f7d-bd81-4c11-8dfb-f11ab86e6398)

**Comment by an-emptymind** [comment]
I have installed it successfully.
Thanks alot for your support.

**Reply by Shiva-Sai-ssb** [reply]
You're welcome. I'm pleased to hear that you've successfully installed Foundry.

**Reply by an-emptymind** [reply]
Let's connect on linkedin:
https://www.linkedin.com/in/raghavendra-havale-464b1b164/

**Reply by Shiva-Sai-ssb** [reply]
Sure, I have sent a connection request.

**Comment by codewizard00** [comment]
thanks a lot bro i was facing the same issue and the above instruction solved my issues 
send you a connection request.
thanks a lot again
